### ISSUE TITLE
is_valid() doesn't work with own Validation Rule

### ISSUE THREAD

**Issue by marcus-at-localhorst** [Wixel/GUMP#121]
Hello,

I have a problem with extending the class and I am not sure if I do something wrong.
This is my code:

``` php
require "gump.class.php";

class MyGUMP extends GUMP
{

    public static function filter_artnum($value, $param = NULL)
    {
        $value = preg_replace('@([^a-zA-Z0-9\-\,|_\. ]+)@si', '', $value );
        if(!is_null($param)){
            $value = mb_substr($value, 0, (int) $param);
        }
        return $value;
    }

    public static function validate_artnum($field, $input, $param = NULL)
    {

        if (!isset($input[$field])) {
            return;
        }
       return (bool) preg_match('@([^a-zA-Z0-9\-\,|_\. ]+)@si', $input[$field]);
    }

} // EOC


$gump = new MyGUMP();

$_REQUEST['quick_artnum'] = 'xyz#';

$x = $gump->is_valid($_REQUEST, ['quick_artnum' => 'artnum'] );
$y = $gump->validate($_REQUEST, ['quick_artnum' => 'artnum'] );
$z = $gump->filter($_REQUEST, ['quick_artnum' => 'artnum'] );

var_dump($x,$y,$z);
exit;
```

While `$y` and `$z` are working/return something, `$x` throws this

```
[12-Jan-2016 18:31:23 Europe/Berlin] PHP Fatal error:  Uncaught exception 'Exception' with message 'Validator method 'validate_artnum' does not exist.' in E:\WEB\xxx\html\template\lib\gump.class.php:405
Stack trace:
#0 E:\WEB\xxx\html\template\lib\gump.class.php(250): GUMP->validate(Array, Array)
#1 E:\WEB\xxx\html\template\lib\gump.class.php(85): GUMP->run(Array)
#2 E:\WEB\xxx\files\actions.inc.php(11): GUMP::is_valid(Array, Array)
#3 E:\WEB\xxx\files\config.inc.php(393): require_once('E:\\WEB\\live.mhm...')
#4 E:\WEB\xxx\html\kategorie.php(3): require_once('E:\\WEB\\live.mhm...')
#5 {main}
  thrown in E:\WEB\xxx\html\template\lib\gump.class.php on line 405
```

I wonder why it's running `GUMP->validate(Array, Array)`and not  `MyGUMP->validate(Array, Array)`

Setting the rules like this, works:

``` php
GUMP::add_validator("artnum", function($field, $input, $param = NULL) {
    return (bool) !preg_match('@([^a-zA-Z0-9\-\,|_\. ]+)@si', $input[$field]);
});

GUMP::add_filter("artnum", function($value, $param = NULL) {
    $value = preg_replace('@([^a-zA-Z0-9\-\,|_\. ]+)@si', '', $value );
    if(!is_null($param)){
        $value = mb_substr($value, 0, (int) $param);
    }
    return $value;
});

GUMP::is_valid($_REQUEST, ['quick_artnum' => 'artnum'] );
```

Any idea?

Thanks, M.

**Comment by filisko** [comment]
Hi, I think that this happens because **is_valid** method uses **get_instance** method and this creates an instance of GUMP itself and not the extended class.

**Comment by filisko** [comment]
Hi @marcus-at-localhorst ,

In a nutshell, this happens because **is_valid** method uses **get_instance** method, which creates an instance of GUMP itself, and not the extended class.

The same problem is on issue #142 by the way.

Just fixed on commit e3344db.

Regards,
Filis.
